### PR TITLE
put *gzip.Writer back to pool

### LINKED ISSUE #1439: gzip.Writer not return to Pool？
https://github.com/grpc/grpc-go/blob/master/rpc_util.go#L70
Is there missing defer to return z to the *gzip.Writer Pool？

### PR DESCRIPTION
**Pull request by Zeymo**
fixes #1439

### CODE REVIEW

File: rpc_util.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ func NewGZIPCompressor() Compressor {
 
 func (c *gzipCompressor) Do(w io.Writer, p []byte) error {
 	z := c.pool.Get().(*gzip.Writer)
+	defer func(){
```
Comment (dfawley) [inline rpc_util.go]: `defer c.pool.Put(z)`

**Comment by dfawley** [comment]
Thanks for finding this and for the fix!

It looks like the second commit was done by xuechen.wzm -- if you redo this commit with your other username and `git push -f`, I believe that should take care of it.

**Comment by Zeymo** [comment]
sry for my careless : (

**Comment by dfawley** [comment]
I'm not sure if this will work.  I thought that since there was a commit by another author, it would say you hadn't signed the CLAs.  But it's not complaining at the moment.

I'll be squashing and merging anyway, and since both authors are you, LGTM and let's hope it actually works.  Thanks again for the fix.

**Pull request merged**